File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        ans = [-1, -1]
        seen = {}

        for idx, num in enumerate(nums):
            diff = target - num
            if diff in seen:
                return [seen[diff], idx]
            seen[num] = idx
        return ans

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        
        Dictionary<int, int> diff = new Dictionary<int, int>();
        
        for (int i = 0; i < nums.Count(); i++) {
            int num = nums[i];
            int difference = target - num;
            if (diff.ContainsKey(difference)) {
                int[] res = new int[2];
                res[0] = diff[difference];
                res[1] = i;
                return res;
            }
            
            diff[num] = i;
        }
        
        return null;
        
        
        
        
        
    }
}